test/dispatch.test.js

```diff
@@ -0,0 +1,130 @@
+import { app } from ".."
+import { deepEqual } from "testmatrix"
+
+const dispatch = action => {
+  let state
+  app({
+    init: action,
+    subscriptions: s => (state = s, [])
+  })
+  return state
+}
```

Comment:
What's up with `s => (state = s, [])`?

Reply:
this allows to `catch and return state` and return any value into subscriptions (it's shorter than `s => (state = s, undefined)`)
* hm, it's possible `s => (state = s,)` to return undefined

Reply:
Haha, nice hack. Now, what would be a more readable way to write this?

Reply:
I believe the most balanced short/readable is existing version `s => (state = s, [])`, but if you worry about comma operator, I can change it to `s => { state = s; return }` or even more readable:
```javascript
s => {
  state = s
  return
}
```

Reply:
@zaceno Thoughts?

Reply:
@mrozbarry @zaceno Do you have an opinion on this?

Reply:
maybe hyperapp should expose internal state variable via api to be able to get current app state at any given time?

Reply:
@frenzzy you can find discussion about exposing app state https://github.com/jorgebucaran/hyperapp/issues/863

Reply:
I don't think that's a bad solution. Initially I thought that using a promise would be better, since it could be more hyperapp-ish, and that looked like:

```js
let resolve = null;
const promise = new Promise((done) => { resolve = done});
const ResolverFX = (dispatch, state) => {
  resolve(state);
  return () => {};
});
const Resolver = state => [ResolverFX, state];

// App with resolver sub

const state = await promise;
```

There's pros and cons to both. The proposed code technically makes the app impure, but it's simpler code. Mine moves the impurity to a sub, but it's obviously more code and less manageable.

That's too say the proposed code should be fine.

Reply:
I have no contrary opinion. Looks fine to me.

Reply:
@jorgebucaran I thought you were worrying about 'comma operator' :smile:  
OK, how about that?
```javascript
const dispatch = (init, view) => 
  app({ init, view })

// and test
{
  test: "test name",
  assert: deepEqual,
  actual: done => 
    dispatch({ some: true }, done),
  expected: { effect: true }
}
```

Reply:
@jorgebucaran dispatch wrapper update (see latest commit)